components/merkledb/src/backends/rocksdb.rs

```diff
@@ -152,6 +153,17 @@ impl Database for RocksDB {
         w_opts.set_sync(true);
         self.do_merge(patch, &w_opts)
     }
+
+    fn create_checkpoint(&self, path: &Path) -> crate::Result<Box<dyn Database>> {
```

Comment:
```suggestion
    fn create_checkpoint<P: AsRef<Path>>(&self, path: P) -> crate::Result<Box<dyn Database>> {
```

Reply:
This will, unfortunately, break object safety for the `Database` trait.

Reply:
And even for `Where: Self`?

Reply:
> And even for where: Self?

This will, somewhat, break the purpose because `create_checkpoint` will be unusable on trait object when database [is only available via trait object][1] most of the time.
Option i see is to extend [`Database` bounds][2] with `Any` to allow downcast but i think it's too much for this.

[1]: https://docs.rs/exonum/0.12.0/src/exonum/blockchain/mod.rs.html#82
[2]: https://docs.rs/exonum-merkledb/0.12.0/src/exonum_merkledb/db.rs.html#418

Reply:
Addressed in new version.